src/Compilers/Core/Portable/Operations/OperationNodes.cs

```diff
@@ -7913,11 +7917,22 @@ public override IOperation Value
     /// </summary>
     internal sealed partial class DeclarationPatternOperation : Operation, IDeclarationPatternOperation
     {
-        public DeclarationPatternOperation(ISymbol declaredSymbol, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
-            base(OperationKind.DeclarationPattern, semanticModel, syntax, type, constantValue, isImplicit)
-        {
+        public DeclarationPatternOperation(
+            ITypeSymbol inputType,
+            ITypeSymbol matchedType,
+            ISymbol declaredSymbol,
+            bool matchesNull,
+            SemanticModel semanticModel,
+            SyntaxNode syntax,
+            bool isImplicit)
+            : base(OperationKind.DeclarationPattern, semanticModel, syntax, type: default, constantValue: default, isImplicit)
+        {
+            InputType = inputType;
+            MatchedType = matchedType;
             DeclaredSymbol = declaredSymbol;
+            MatchesNull = matchesNull;
         }
+        public bool MatchesNull { get; }
```

Comment:
>MatchesNull [](start = 20, length = 11)

Please align the name between operation and test output. `AcceptsNull` or `MatchesNull`.